src/pdc-api.ts

```diff
@@ -0,0 +1,100 @@
+import { client } from './client';
+import type { AccessTokenSet } from './oidc';
+
+const callPdcApi = async <T>(
+  baseUrl: string,
+  path: string,
+  params: Record<string, string>,
+  token: AccessTokenSet,
+  method: 'get' | 'post',
+  data?: unknown,
+): Promise<T> => {
+  const url = new URL(path, baseUrl);
+  url.search = new URLSearchParams(params).toString();
+  const response = await client.request<T>(
+    {
+      method,
+      url: url.toString(),
+      headers: {
+        authorization: `Bearer ${token.access_token}`,
+      },
+      data,
+    },
+  );
+  return response.data;
+};
+
+interface ApiBaseField {
```

Comment:
I would prefer to omit the `Api`. Alternatives could be `BaseField` or `PdcBaseField` or `PDCBaseField`. I wouldn't even mention it except that this code appears intended to be common code for multiple scripts in future.

Reply:
This relates to a discussion we started in https://github.com/PhilanthropyDataCommons/data-viewer/issues/108 !

In particular, we found it useful to transform the data structures returned by the API to other client-specific data structures, and ended up having a hard time coming up with names for those data structures. These definitions were copied from the data-viewer, but I think the same constraints may apply - although probably not as strongly. Additionally, I think there's some merit in being consistent between these two client repos.

Calling it just "BaseField" gives these data structures a priority that might not be appropriate, does not communicate their origin, and makes search (or search-and-replace) more difficult. The `PDC` prefix doesn't suggest to me a distinction between API-facing and internal-facing data structures.

What do you think with all that additional context, @bickelj?

Reply:
I think it's good enough. My other thoughts can go in https://github.com/PhilanthropyDataCommons/service/issues/408.